testing/async/test_policy_overloads.h

```diff
@@ -0,0 +1,421 @@
+      std::tuple_element_t<postfix_idx, postfix_args_type>;
+    postfix_tuple_type const postfix_tuple = get_postfix_tuple<postfix_idx>();
+
+    // Generate index sequences for the tuples:
+    constexpr auto prefix_tuple_size  = std::tuple_size<prefix_tuple_type>{};
+    constexpr auto postfix_tuple_size = std::tuple_size<postfix_tuple_type>{};
+    using prefix_index_seq  = std::make_index_sequence<prefix_tuple_size>;
+    using postfix_index_seq = std::make_index_sequence<postfix_tuple_size>;
+
+    // Use unique, non-const inputs for each invocation to support in-place
+    // algo_def configurations.
+    input_type input_a   = algo_def::generate_input(num_values);
+    input_type input_b   = algo_def::generate_input(num_values);
+    input_type input_c   = algo_def::generate_input(num_values);
+    input_type input_d   = algo_def::generate_input(num_values);
+    input_type input_ref = algo_def::generate_input(num_values);
+
+    output_type output_a   = algo_def::generate_output(num_values, input_a);
+    output_type output_b   = algo_def::generate_output(num_values, input_b);
+    output_type output_c   = algo_def::generate_output(num_values, input_c);
+    output_type output_d   = algo_def::generate_output(num_values, input_d);
+    output_type output_ref = algo_def::generate_output(num_values, input_ref);
+
+    // Invoke multiple overlapping async algorithms, capturing their outputs
+    // and events/futures:
+    auto e_a = algo_def::invoke_async(prefix_tuple,
+                                      prefix_index_seq{},
+                                      input_a,
+                                      output_a,
+                                      postfix_tuple,
+                                      postfix_index_seq{});
+    auto e_b = algo_def::invoke_async(prefix_tuple,
+                                      prefix_index_seq{},
+                                      input_b,
+                                      output_b,
+                                      postfix_tuple,
+                                      postfix_index_seq{});
+    auto e_c = algo_def::invoke_async(prefix_tuple,
+                                      prefix_index_seq{},
+                                      input_c,
+                                      output_c,
+                                      postfix_tuple,
+                                      postfix_index_seq{});
+    auto e_d = algo_def::invoke_async(prefix_tuple,
+                                      prefix_index_seq{},
+                                      input_d,
+                                      output_d,
+                                      postfix_tuple,
+                                      postfix_index_seq{});
+
+    // Let reference calc overlap with async testing:
+    algo_def::invoke_reference(input_ref,
+                               output_ref,
+                               postfix_tuple,
+                               postfix_index_seq{});
+
```

Comment:
We should probably leave a comment here explaining that waiting happens in `compare_outputs`

Reply:
Done.